test/js/unit/Utils.js

```diff
@@ -91,24 +91,59 @@
             deepEqual(
                 result,
                 {
+                    protocol: "http:",
+                    host: "tincanapi.com:8080",
+                    hostname: "tincanapi.com",
+                    port: "8080",
+                    pathname: "/TinCanJS/Test/TinCan.Utils_parseURL/test",
+                    search: "",
+                    hash: "",
                     params: {},
                     path: "http://tincanapi.com:8080/TinCanJS/Test/TinCan.Utils_parseURL/test"
                 },
-                 "return value: no params"
+                "return value: no params"
             );
 
             result = TinCan.Utils.parseURL("http://tincanapi.com:8080/TinCanJS/Test/TinCan.Utils_parseURL/test?paramA=1&paramB=2");
             deepEqual(
                 result,
                 {
+                    protocol: "http:",
+                    host: "tincanapi.com:8080",
+                    hostname: "tincanapi.com",
+                    port: "8080",
+                    pathname: "/TinCanJS/Test/TinCan.Utils_parseURL/test",
+                    search: "?paramA=1&paramB=2",
+                    hash: "",
                     params: {
                         paramA: "1",
                         paramB: "2"
                     },
                     path: "http://tincanapi.com:8080/TinCanJS/Test/TinCan.Utils_parseURL/test"
                 },
-                 "return value: with params"
-             );
+                "return value: with params"
+            );
+
+            result = TinCan.Utils.parseURL("https://tincanapi.com/TinCanJS/Test/TinCan.Utils_parseURL/test?paramA=1&paramB=2&weirdParam=odd?secondQuestionMark#withHash");
+            deepEqual(
+                result,
+                {
+                    protocol: "https:",
+                    host: "tincanapi.com",
+                    hostname: "tincanapi.com",
+                    port: undefined,
+                    pathname: "/TinCanJS/Test/TinCan.Utils_parseURL/test",
+                    search: "?paramA=1&paramB=2&weirdParam=odd?secondQuestionMark",
+                    hash: "#withHash",
+                    params: {
```

Comment:
Note that prior to building, this test was returning no params, which shows this PR fixed that bug.